src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/AnalysisEntityDataFlowOperationVisitor.cs

```diff
@@ -87,10 +87,13 @@ protected void ResetAnalysisData(DictionaryAnalysisData<AnalysisEntity, TAbstrac
         {
             // Reset the current analysis data, while ensuring that we don't violate the monotonicity, i.e. we cannot remove any existing key from currentAnalysisData.
             // Just set the values for existing keys to ValueDomain.UnknownOrMayBeValue.
-            var keys = currentAnalysisDataOpt?.Keys.ToImmutableArray();
-            foreach (var key in keys)
+            if (currentAnalysisDataOpt != null)
```

Comment:
💡  If you aren't going to mark the parameter as allowing null, you might as well just get rid of the null check and the `Opt` suffix. 😄